src/components/UserManagement/SetUpFinalDayPopUp.test.jsx

```diff
@@ -0,0 +1,71 @@
+import { render, fireEvent, screen, waitFor } from '@testing-library/react';
+import '@testing-library/jest-dom/extend-expect';
+import moment from 'moment';
+import SetUpFinalDayPopUp from './SetUpFinalDayPopUp';
+
+describe('SetUpFinalDayPopUp Component', () => {
+  test('renders the component with initial state', () => {
+    render(<SetUpFinalDayPopUp open={true} onClose={() => {}} onSave={() => {}} />);
+    
+    // Ensure the date input is rendered
+    expect(screen.getByTestId('date-input')).toBeInTheDocument();
+    
+    // Ensure the Save and Close buttons are rendered
+    expect(screen.getByText('Save')).toBeInTheDocument();
+    expect(screen.getByText('Close')).toBeInTheDocument();
+  });
+  
+  test('updates the date when input changes', () => {
+    render(<SetUpFinalDayPopUp open={true} onClose={() => {}} onSave={() => {}} />);
+    
+    const dateInput = screen.getByTestId('date-input');
+    
+    // Simulate user entering a date
+    fireEvent.change(dateInput, { target: { value: '2023-12-31' } });
+    
+    // Check if the state is updated
+    expect(dateInput.value).toBe('2023-12-31');
+  });
+
+  test('calls onSave when Save is clicked with a future date', async () => {
+    const onSaveMock = jest.fn();
+    render(<SetUpFinalDayPopUp open={true} onClose={() => {}} onSave={onSaveMock} />);
+    
+    // Simulate user entering a future date
+    fireEvent.change(screen.getByTestId('date-input'), { target: { value: '2023-12-31' } });
```

Comment:
This test failed because `2023-12-31` is not a future date now, you'd better use date like 2100-01-01 instead of a close date.